tc/sgx/trusted_worker_manager/enclave/work_order_processor_wpe.cpp

```diff
@@ -0,0 +1,152 @@
+     * work order key info json
+     *
+     * @param enclave_data - Instance of EnclaveData class
+     * @param wo_req_json_val - Work order request json value
+     */
+    void WorkOrderProcessorWPE::DecryptWorkOrderKeys(
+        EnclaveData* enclave_data, const JsonValue& wo_req_json_val) {
+        
+        // Decrypt Encryption key
+        ByteArray encrypted_session_key_bytes = \
+            HexStringToBinary(encrypted_session_key);
+
+        // Use keys from pre-processing json structure
+        // passed as ext_work_order_data
+        tcf_err_t status = \
+            wo_pre_proc_keys.ProcessPreProcessedWorkOrderKeys(
+                this->ext_work_order_data, enclave_data);
+        tcf::error::ThrowIf<tcf::error::ValueError>(status != TCF_SUCCESS,
+            "Failed to process pre-processed work order keys");
+
+        // Decrypt work order encrypted session key using symmetric key
+        // from pre-processing json
+        session_key = wo_pre_proc_keys.work_order_session_key;
+        ByteArray session_key_iv_bytes = HexStringToBinary(session_key_iv);
+
+        JSON_Object* request_object = json_value_get_object(wo_req_json_val);
+        tcf::error::ThrowIfNull(request_object,
+            "Missing JSON object in work order request");
+
+        JSON_Object* params_object = json_object_dotget_object(request_object, "params");
+        tcf::error::ThrowIfNull(params_object,
+            "Missing params object in work order request");
+
+	JSON_Array* data_array = json_object_get_array(
+            params_object, "inData");
+        size_t count = json_array_get_count(data_array);
+        tcf::error::ThrowIf<tcf::error::ValueError>(count == 0,
+            "Indata is empty");
+
+        size_t i;
+        for (i = 0; i < count; i++) {
+            JSON_Object* data_object = json_array_get_object(data_array, i);
+            WorkOrderDataHandler wo_data(session_key, session_key_iv_bytes);
+            // Use in-data key from preprocessed json
+            wo_data.Unpack(enclave_data, data_object,
+                wo_pre_proc_keys.in_data_keys[i].decrypted_data);
+            data_items_in.emplace_back(wo_data);
+        }
+        data_array = json_object_get_array(params_object, "outData");
+
+        count = json_array_get_count(data_array);
+        for (i = 0; i < count; i++) {
+            JSON_Object* data_object = json_array_get_object(data_array, i);
+            WorkOrderDataHandler wo_data(session_key, session_key_iv_bytes);
+            // Use out-data key from preprocessed json
+            wo_data.Unpack(enclave_data, data_object,
+                wo_pre_proc_keys.out_data_keys[i].decrypted_data);
+            data_items_out.emplace_back(wo_data);
+        }
+    }
+
+    /*
+     * Creates Work order response in json format
+     */
+    JsonValue WorkOrderProcessorWPE::CreateJsonOutput() {
+        JsonValue response = WorkOrderProcessor::CreateJsonOutput();
```

Comment:
Explicitly add a commemt. Calling base class function to create JsonOutput

Reply:
Ok